### PR TITLE
simplify log printing in localstack start command

### PR DESCRIPTION
**Pull request by thrau**
## Motivation

An enterprise user noticed that when using `EXTENSION_DEV_MODE=1` is set, the information is not printed correctly to the logs when using `localstack start`.
I couldn't really find the root cause, but it seems to have something to do with `echo` calls in the docker-entrypoint.sh not showing up in the default outfile, which is the mechanism we use to dump the docker logs.
We then listen on those logs using a `FileListener`.

That seemed a bit roundabout to me now that we have `stream_container_logs`, so I removed the `FileListener` logic alltogether, and just wait for the container to start and then start printing the logs from there.
Since `docker logs -f` prints all the previous logs, it's a pretty robust solution.

I also found an issue where we didn't pass the `Container` instance to the `LocalstackContainerServer`.

## Changes

* Remove `FileListener` and replace it with `ContainerLogPrinter`
* Change the bootstrap logic of `localstack start` accordingly
* Update `LocalstackContainerServer` to also take a `Container` instance

### CODE REVIEW

**Review by simonrw — COMMENTED** [review]
Does this mean we can remove `default_run_outfile` as well?

**Comment by thrau** [comment]
@simonrw i think i would argue to keep it since it keeps the log file of the last run around in the localstack volume which may be useful. since we remove the container there would be no way of inspect logs after it shuts down

**Review by simonrw — APPROVED** [review]
Nice set of changes, and the implementation seems to work nicely

File: localstack/utils/bootstrap.py [hunk 1]
```diff
@@ -907,8 +906,40 @@ def stopped(self) -> Container:
...
+
+        self._closed = threading.Event()
+        self._stream: Optional[CancellableStream] = None
+
+    def _can_start_streaming(self):
+        if self._closed.is_set():
+            raise IOError("Already stopped")
+        if not self.container.running_container:
+            return False
+        return self.container.running_container.is_running()
```
[the remark this answers is not in the document]
Comment (thrau) [reply localstack/utils/bootstrap.py:922]: I suppose because I want to be able to call this before the container is running

**Pull request merged**